keytool/src/main/scala/org/constellation/keytool/DeterministicKeyGen.scala

```diff
@@ -0,0 +1,43 @@
+package org.constellation.keytool
+
+import com.typesafe.scalalogging.StrictLogging
+
+import java.nio.charset.Charset
+import java.security.spec.ECGenParameterSpec
+import java.security.{SecureRandom, _}
+
+object DeterministicKeyGen extends StrictLogging {
```

Comment:
what you think about making it a part of key utils? there are functions for address generation so imo it can be next to it

Reply:
I'll move this to wallet next to bitcoinj code, realized it's probably better there.

Reply:
well, if it's a key gen, it should be in keytool imo. if it's for transaction generation, it should be in wallet.
keytool is a separate module that is injected into both node and wallet, not to repeat the code.

Reply:
Realized what you meant by earlier comment. IMO we should move all the BIP44 code into KeyUtils. Ideally the nodes should all have their own seed instead of a single key. There's a lot of discussion around what the right way to deal with key rotations is here, but at the very least nodes should be able to rotate within a deterministic seed. 

Long term, we'd really need some kind of way to handle key rotations even across seeds. What I was going here for in this PR is a way to generate hash offsets which generate mnemonics per hash offset. 

I.e. goal here is we have a single starting input seed which is kept secret -- we use hash offsets of that master seed to populate a large number of different seeds. Each node is given one of those hash offset seeds at a time. If a node needs to rotate keys, it simply gets a new hash offset copied to it off the original master input (kept off of the node itself.) 

That way, we can take a single master seed and generate a Merkle Proof that determines the proper key rotation sequence based on a master. (I.e. seed_hash_offset_0 -> 1000 addresses, seed_hash_offset_1 -> 1000 different addresses,) build that into a Merkle tree, use the root hash as the PeerId. Then when peer rotates between keys, they provide a proof showing they own it. 

This allows the node hardware to be compromised and key rotation easy. Key rotation protocol across multiple master seeds is much harder, and requires some kind of consensus / recovery procedure. (BLS / Shamir / Multi-sig).

I don't want to cause a classpath conflict by dealing with version upgrade in this PR, just leaving this here to demonstrate fix for BIP-44 and the master seed generation process with hash offsets.

Reply:
Other noteworthy usefulness of doing this is that you don't need to deal with copying keys between local and nodes. As long as they use the same master seed with adjustable hash offset, you can replicate all node keys locally.